### PR TITLE
email support on import fail

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**

### CODE REVIEW

File: apps/bamboohr/email.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from io import StringIO
+
+import sendgrid
+from sendgrid.helpers.mail import Mail, Attachment
+from python_http_client.exceptions import HTTPError
+import base64
+
+from admin_settings import settings
+
+def send_failure_notification_email(employees, number_of_employees ,admin_email):
```
Comment (ashwin1111) [inline apps/bamboohr/email.py:11]: add type, doc string etc

File: fyle_employee_imports/base.py [hunk 2]
```diff
@@ -78,6 +82,8 @@ def get_employee_and_approver_payload(self, hrms_employees):
         employee_emails: List[str] = []
         approver_emails: List[str] = []
         employee_approver_payload: List[Dict] = []
+        unimported_employees: List = []
+        no_unimported_employees: int = 0
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:86]: incomplete_employee_count

File: fyle_employee_imports/base.py [hunk 3]
```diff
@@ -78,6 +82,8 @@ def get_employee_and_approver_payload(self, hrms_employees):
         employee_emails: List[str] = []
         approver_emails: List[str] = []
         employee_approver_payload: List[Dict] = []
+        unimported_employees: List = []
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:85]: incomplete_employees

File: fyle_employee_imports/base.py [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
+from django.conf import settings
+
 class FyleEmployeeImport():
 
     def __init__(self, org_id: int, user):
         self.org_id = org_id
         self.user = user
-        self.bamboohr = BambooHr.objects.get(org_id__in=self.org_id)
+        self.bamboohr = BambooHr.objects.get(org_id__in=[self.org_id])
+        self.bamboohr_configuration = BambooHrConfiguration.objects.get(org_id__in=[self.org_id])
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:19]: move all bamboo specific things to bamboo class plis

File: fyle_employee_imports/base.py [hunk 5]
```diff
@@ -97,6 +103,12 @@ def get_employee_and_approver_payload(self, hrms_employees):
                         'approver_emails': employee.detail['approver_emails']
                     })
                     approver_emails.extend(employee.detail['approver_emails'])
+            else:
+                no_unimported_employees += 1
+                unimported_employees.append({'name': employee.detail['full_name'], 'id':employee.destination_id})
+        
+        admin_email = self.bamboohr_configuration.additional_email_options['admin_email']
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:110]: get this from bamboo class. Similar to how we sync.

File: fyle_employee_imports/base.py [hunk 6]
```diff
@@ -131,27 +129,39 @@ def fyle_employee_import(self, hrms_employees):
...
+            self.set_employee_exported_at()
 
         if employee_approver_payload:
             self.platform_connection.bulk_post_employees(employees_payload=employee_approver_payload)
             
-            self.bamboohr.employee_exported_at = datetime.now()
+            self.set_employee_exported_at()
         
-        self.bamboohr.save()
+        self.save_hrms()
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:139]: save_employee_exported_at_time()

**Review by ashwin1111 — APPROVED** [review]

**Review by NileshPant1999 — APPROVED** [review]

**Pull request merged**